### PR TITLE
Client provided layer size information

### PR DESCRIPTION
**Pull request by davidzhao**

### CODE REVIEW

File: livekit_models.proto [hunk 1]
```diff
@@ -73,6 +73,23 @@ message TrackInfo {
...
+
+enum VideoQuality {
+  LOW = 0;
+  MEDIUM = 1;
+  HIGH = 2;
+}
+
+// provide information about available spatial layers
+message VideoLayer {
+  // for non-simulcast or SVC, this should be HIGH
```
Comment (boks1971) [inline livekit_models.proto:87]: I am thinking SVC should report configured layers too. Not sure though if there is any event on encoder internally changing dimensions. Maybe, for now leave out comment about SVC having to use HIGH?

**Review by boks1971 — APPROVED** [review]

**Comment by davidzhao** [comment]
@hiroshihorie do you think this could work for the dynamic-dimension feature that you had planned?
@frostbyte73 we are adding another field to TrackInfo, wanted to consider how this data is synchronized within cloud.

File: livekit_models.proto (hunk 1 above)
Comment (davidzhao) [reply livekit_models.proto]: yes, I meant to type non-(simulcast or SVC), but it came out wrong.. lol

**Comment by davidzhao** [comment]
> Dimensions information will be required per layer ?

yes, this allows you to provide custom layer sizes when simulcasting.

**Pull request merged**